### PR TITLE
add FreDF loss combining time-domain MSE and frequency-domain MAE

### LINKED ISSUE #1301: [<Library component: Models|Core|etc...>] Recommend a versatile and easy-to-implement loss function
### Description

Hi there,

I am writing to introduce a loss function tailored for time-series forecasting proposed in [FreDF](https://openreview.net/forum?id=4A9IdSa1ul). It can be easily implemented through two lines of code.

```python
# The canonical temporal loss
loss_tmp = ((outputs-batch_y)**2).mean()
# The proposed frequency loss
loss_feq = (torch.fft.rfft(outputs, dim=1) - torch.fft.rfft(batch_y, dim=1)).abs().mean() 
# Note. The frequency loss can be used individually or fused with the temporal loss using finetuned relative weights. Both witness performance gains, see the ablation study in our paper.

``` 

The efficacy of the loss function has been investigated through extensive experiments in [this study](https://openreview.net/forum?id=4A9IdSa1ul). I recognized that there is a `losses` module in NeuralForecast so I guess it would be helpful to include this plain and effective method.

### Use case

_No response_

### PR DESCRIPTION
**Pull request by DresdenGman**
Following up on #1301 — implements the FreDF loss proposed by Lin et al. (2023).

FreDF combines time-domain MSE with frequency-domain MAE via torch.fft.rfft, weighted by a tunable alpha parameter (default 0.5). When alpha=0 it reduces to pure MSE; when alpha=1 it reduces to pure frequency-domain MAE.

Added FreDF to __all__ in losses/pytorch.py, inheriting BasePointLoss to stay consistent with the existing loss API.

Closes #1301

### CODE REVIEW

**Comment by marcopeix** [comment]
@DresdenGman thanks for the contribution! Could you add a few tests in `neuralforecast/tests/test_losses/test_pytorch.py` please. Especially for when alpha is 0 and 1, that will validate the implementation is sound. Thanks!

**Comment by DresdenGman** [comment]
Added three tests in `test_pytorch.py`: alpha=0 (reduces to MSE), alpha=1 (reduces to frequency-domain MAE), and perfect forecast gives zero loss for any alpha.

File: neuralforecast/tests/test_losses/test_pytorch.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+    # No duplicates should produce no warning
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        check = MQLoss(level=[80, 90])
+        assert len(w) == 0
+    assert len(check.quantiles) == 5
+
+
+def test_fredf_alpha0_equals_mse():
```
Comment (marcopeix) [inline neuralforecast/tests/test_losses/test_pytorch.py]: The tests are good, but it seems like the entire file is new. That's not right. You should modify the existing file: https://github.com/Nixtla/neuralforecast/blob/main/tests/test_losses/test_pytorch.py

**Review by marcopeix — CHANGES_REQUESTED** [review]
@DresdenGman, see my comment. Seems like you're adding an entirely new test file, but you should modify the existing test file.

**Comment by DresdenGman** [comment]
Fixed — moved the FreDF tests into the existing tests/test_losses/test_pytorch.py file and removed the accidental duplicate test path.

The loss tests pass locally: 15 passed.

**Review by marcopeix — APPROVED** [review]
Thanks for the contribution!
